pandas/core/_algos/__init__.py

```diff
@@ -0,0 +1,9 @@
+"""
+core._algos is for algorithms that operate on ndarray and ExtensionArray.
+These should:
+
+- Assume that any Index, Series, or DataFrame objects have already been unwrapped.
+- Assume that any list arguments have already been cast to ndarray/EA.
+- Not depend on Index, Series, or DataFrame, nor import any of these.
+- May dispatch to ExtensionArray methods, but should not import from core.arrays.
```

Comment:
I don't know what else you are planning to put here, but it could also be pure-numpy algos? That would also have value in being very clear. 
Eg the `shift` you moved here now is strictly numpy arrays.

Reply:
Yah, I have no problem with some functions in here being explicitly ndarray-only.

Reply:
I wasn't speaking about some functions, but the full file. 

But so, what other functions do you envision to put here? (as for a single function it's not worth creating a module I think)

Reply:
> I wasn't speaking about some functions, but the full file.

That may end up being a reasonable organization, but it isn't obvious to me.  e.g. if we end up with a `shift` that dispatches to `shift_ea` and `shift_ndarray`, would we want those to all be in the same file or split across multiple files?